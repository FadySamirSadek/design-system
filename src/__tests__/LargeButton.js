import React from 'react'
import renderer from 'react-test-renderer'
import { LargeButton, theme } from '..'

describe('LargeButton', () => {
  test('renders', () => {
    const json = renderer.create(<LargeButton />).toJSON()
    expect(json).toMatchSnapshot()
  })

  test('larger', () => {
    const json = renderer.create(<LargeButton bg="accent" />).toJSON()
    expect(json).toMatchSnapshot()
    expect(json).toHaveStyleRule('background-color', theme.colors.accent)
    expect(json).toHaveStyleRule('color', theme.colors.white)
    expect(json).toHaveStyleRule('padding-left', theme.space[4] + 'px')
  })
})
