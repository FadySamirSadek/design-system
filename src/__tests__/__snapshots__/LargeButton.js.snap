// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LargeButton larger 1`] = `
.c0 {
  padding-left: 32px;
  padding-right: 32px;
  padding-top: 16px;
  padding-bottom: 16px;
  background-color: #7280ed;
  color: #ffffff;
  font-size: 16px;
  -webkit-font-smoothing: antialiased;
  display: inline-block;
  vertical-align: middle;
  text-align: center;
  text-decoration: none;
  font-family: inherit;
  font-weight: 700;
  line-height: 1.125;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  cursor: pointer;
  -webkit-transition: .125s box-shadow ease-out;
  transition: .125s box-shadow ease-out;
  box-shadow: 0 2px 12px rgba(0,0,0,0.125);
  border-radius: 4px;
  border-width: 0;
  border-style: solid;
}

.c0:hover,
.c0:focus {
  outline: 0;
  box-shadow: 0 2px 12px 2px rgba(0,0,0,0.125);
}

.c0:active {
  outline: 0;
  box-shadow: 0 4px 16px 2px rgba(0,0,0,0.125);
}

@media screen and (min-width:32em) {
  .c0 {
    font-size: 20px;
  }
}

<a
  className="c0"
  color="white"
/>
`;

exports[`LargeButton renders 1`] = `
.c0 {
  padding-left: 32px;
  padding-right: 32px;
  padding-top: 16px;
  padding-bottom: 16px;
  background-color: #e42d42;
  color: #ffffff;
  font-size: 16px;
  -webkit-font-smoothing: antialiased;
  display: inline-block;
  vertical-align: middle;
  text-align: center;
  text-decoration: none;
  font-family: inherit;
  font-weight: 700;
  line-height: 1.125;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  cursor: pointer;
  -webkit-transition: .125s box-shadow ease-out;
  transition: .125s box-shadow ease-out;
  box-shadow: 0 2px 12px rgba(0,0,0,0.125);
  border-radius: 4px;
  border-width: 0;
  border-style: solid;
}

.c0:hover,
.c0:focus {
  outline: 0;
  box-shadow: 0 2px 12px 2px rgba(228,45,66,.25);
}

.c0:active {
  outline: 0;
  box-shadow: 0 4px 16px 2px rgba(228,45,66,.375);
}

@media screen and (min-width:32em) {
  .c0 {
    font-size: 20px;
  }
}

<a
  className="c0"
  color="white"
/>
`;
